spec/routes/api/cli/help_spec.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require_relative "spec_helper"
+
+RSpec.describe Clover, "cli help" do
+  it "shows help for top level command with no arguments" do
+    expect(cli(%w[help])).to eq <<~OUTPUT
+      Usage: ubi [options] [subcommand [subcommand-options] ...]
+
+      Options:
+              --version                    show program version
+              --help                       show program help
+              --confirm=confirmation       confirmation value (not for direct use)
+
+      Subcommands: help pg vm
+    OUTPUT
+  end
+
+  it "shows help for specific command if given" do
+    expect(cli(%w[help help])).to eq <<~OUTPUT
+      Usage: ubi help [options] [command [subcommand]]
+
+      Options:
+          -r, --recursive                  also show documentation for all subcommands of command
+          -u, --usage                      only show usage
+    OUTPUT
+  end
+
+  it "shows help for specific subcommand if given" do
+    expect(cli(%w[help vm ssh])).to eq <<~OUTPUT
+      Usage: ubi vm location/(vm-name|_vm-ubid) [options] ssh [ssh-options --] [remote-cmd [remote-cmd-arg ...]]
+    OUTPUT
+  end
+
+  it "shows only usage if the -u flag is given" do
+    expect(cli(%w[help -u help])).to eq <<~OUTPUT
+      Usage: ubi help [options] [command [subcommand]]
+    OUTPUT
+  end
+
+  it "shows help for all subcommands of command if -r is given" do
+    expect(cli(%w[help -r vm])).to include <<~OUTPUT
+      Usage: ubi vm location/vm_name create [options] public_key
+
+      Options:
+          -6, --ipv6-only                  do not enable IPv4
+    OUTPUT
+  end
+
+  it "shows usage for all subcommands of command if -ru is given" do
+    expect(cli(%w[help -ru vm])).to include <<~OUTPUT
+      Usage: ubi vm list [options]
+      Usage: ubi vm location/vm_name create [options] public_key
+    OUTPUT
+  end
+
+  it "shows error and help for top level if top level command invalid" do
+    expect(cli(%w[help foo], status: 400)).to eq <<~OUTPUT
+      invalid command: foo
```

Comment:
It might look more consistent if `invalid command` were titlecase, or at least sentence case. Other things before colons are, so it seems a little inconsistent. That might also draw the eye. You could also consider additional treatment for errors or warnings, in my recent usage I've used a exclamation prefix, so something like `! Invalid Command: foo` might stick out more and hopefully draw the eye more as well.

Reply:
That makes sense.  I'll make this change and add it to this PR.

Reply:
Pushed a commit for this.